samcli/commands/init/init_templates.py

```diff
@@ -179,7 +181,11 @@ def get_preprocessed_manifest(
         # at the top of list template example displayed to the Customer.
         preprocessed_manifest = {"Hello World Example": {}}  # type: dict
         for template_runtime in manifest_body:
-            if filter_value and filter_value != template_runtime:
+            if (
+                is_custom_runtime(filter_value)
+                and filter_value != get_custom_runtime_base_runtime(template_runtime)
+                or (filter_value and not is_custom_runtime(filter_value) and filter_value != template_runtime)
+            ):
```

Comment:
The condition of this if statement seems too complicated and hard to maintain, could we break it down into multiple if statements and document each cases of these conditions?